identity/web.py

```diff
@@ -248,8 +256,8 @@ def log_out(self, homepage):
         """
         self._session.pop(self._USER, None)  # Must
         self._session.pop(self._TOKEN_CACHE, None)  # Optional
-        return "{authority}/oauth2/v2.0/logout?post_logout_redirect_uri={hp}".format(
-            authority=self._authority, hp=homepage)
+        e = self._get_oidc_config().get("end_session_endpoint")
```

Comment:
> Is there a scenario or possibility for this endpoint to be unavalable? Its probably negligible but ...

Yes, it is totally possible. No, it is not negligible. That is why the next line contains a safety check.

> if it can [be unavailable] then would that mean that the users can't logout ?

Indeed but in that case, the user will still be able to logout from the current app, because this implementation cleared the local session in the previous two lines.

Reply:
I tried out the scenario of clearing the cache, whereas that will be cleared, it won't redirect to the page it needs to redirect to after completing the logout so they'll most likely end up with a 404.

Maybe this isn't a concern for now

Reply:
> I **tried out** the scenario of clearing the cache, ... so **they'll most likely** end up with a 404.

Our offline conversation agreed on having a reproducible test result to know whether it will end up with a 404, and then we go from there.